lib/common/value-utils.js

```diff
@@ -0,0 +1,44 @@
+'use strict'
+
+const { nativeImage, NativeImage } = process.atomBinding('native_image')
+
+const objectMap = function (object, mapper) {
+  return Object.fromEntries(Object.entries(object).map(([ key, value ]) => [ key, mapper(value) ]))
```

Comment:
I know this was discussed in a thread below but I really don't like this, we should be playing [Code Golf](https://en.wikipedia.org/wiki/Code_golf) with our internal code.

I should be able to look at this function and **instantly** see what it does.  The sample that @ckerr wrote IMO is much more readable --> https://github.com/electron/electron/pull/17200/files#r261829857

Reply:
If `Object.fromEntries()` is desirable -- e.g., if it has some low-level v8 code that makes object-building faster than the code I suggested -- then we can dial the knob a little and have something performant but still readable:

```js
const objectMap = (mapper, source) => {
  const sourceEntries = Object.entries(source)
  const targetEntries = sourceEntries.map(([key, val]) => [key, mapper(val)])
  return Object.fromEntries(targetEntries)
}
```

Reply:
...so I went to benchmark this and see if `Object.fromEntries()` contains any intrinsic advantage, and I'm seeing this:

```js
console.log({ 'process.version': process.version })

const mapObjectFromEntries = (mapper, source) => {
  const sourceEntries = Object.entries(source)
  const targetEntries = sourceEntries.map(([key, val]) => [key, mapper(val)])
  return Object.fromEntries(targetEntries)
}
```

yields:

```txt
{ 'process.version': 'v12.0.0-nightly20190302453ed05343' }
/tmp/foo.js:17
  return Object.fromEntries(targetEntries)
                ^

TypeError: Object.fromEntries is not a function
    at mapObjectFromEntries (/tmp/foo.js:17:17)
```

It appears that `Object.fromEntries()` is still bleeding-edge. Are we certain that this code actually works in Electron? And if not, how are all the tests passing -- are these paths covered in the specs?

Reply:
It does work.

Reply:
I can defunitel use the code from @ckerr if its more readable

Reply:
> It appears that Object.fromEntries() is still bleeding-edge. Are we certain that this code actually works in Electron?

@ckerr It was introduced in Chromium 73 behind a flag (enabled by default) https://www.chromestatus.com/feature/5747878282657792